examples/lightningsynth.py

```diff
@@ -1,29 +1,96 @@
-#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
-# # lightningsynth
-#
-# Profiling for our synth on GPUs
-#
-# If this hasn't been merged to master yet, run:
-# ```
-# # !pip uninstall -y torchsynth
-# # !pip install git+https://github.com/turian/torchsynth.git@lightning-synth
-# ```
+"""lightningsynth.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/github/turian/torchsynth/blob/lightning-synth/examples/lightningsynth.ipynb
+
+# lightningsynth
+
+Profiling for our synth on GPUs
+
+Make sure you are on GPU runtime
+
+If this hasn't been merged to master yet, run:
+```
+!pip uninstall -y torchsynth
+!pip install git+https://github.com/turian/torchsynth.git@lightning-synth
+```
+"""
+
+#!pip uninstall -y torchsynth
+#!pip install git+https://github.com/turian/torchsynth.git@lightning-synth
+
+#!pip install torchvision
 
 import torch
 import torch.tensor as T
 from tqdm.auto import tqdm
 
+import torch
+
+# import torchvision.models as models
+import torch.autograd.profiler as profiler
+import pytorch_lightning as pl
+
 from torchsynth.globals import SynthGlobals
 from torchsynth.synth import Voice
+import torchsynth.module
+
+gpus = torch.cuda.device_count()
+print("Usings %d gpus" % gpus)
 
-synthglobals = SynthGlobals(batch_size=T(256))
+# Note this is the batch size for our synth!
+# i.e. this many synth sounds are generated at once
+# Not the batch size of the datasets
+BATCH_SIZE = 256
+
+import multiprocessing
+
+ncores = multiprocessing.cpu_count()
+print(f"Using ncores {ncores} for generating batch numbers (low CPU usage)")
+
+
+class batch_idx_dataset(torch.utils.data.Dataset):
+    def __init__(self, num_batches):
+        self.num_batches = num_batches
+
+    def __getitem__(self, idx):
+        return idx
+
+    def __len__(self):
+        return self.num_batches
+
+
+synth1B = batch_idx_dataset(1024 * 1024 * 1024 // BATCH_SIZE)
```

Comment:
Do we want this to be something a bit closer to a billion? I guess it's not a big issue but this is 73M larger than 1B.